typespec-extension/src/code-model-builder.ts

```diff
@@ -2689,13 +2368,13 @@ export class CodeModelBuilder {
     return getWireName(this.sdkContext, target);
   }
 
-  private isReadOnly(target: ModelProperty): boolean {
+  private isReadOnly(target: ModelProperty | undefined): boolean {
```

Comment:
This is a problem brought up before, we checked segment in `isReadOnly`, but TCGC does not handle segment info.
So I keep this method..

https://github.com/Azure/autorest.java/blob/3491b346ae7b6e6cac92417ee59cd1b6dbe76c86/typespec-extension/src/code-model-builder.ts#L2371-L2390

Reply:
If this does not affect test or service, we can use theirs.

If something wrong in future, create issue for TCGC.

Reply:
Also, TCGC have visibility, use that https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L370

Reply:
I did not modify the two functions (`isSecret`, `isReadOnly`) because there is difference between TCGC's visibility and TSP compiler's. After chatting offline, our agreement is to ignore the difference and modify the two functions using TCGC types. If our our tests can pass, we can keep using TCGC's one.